### ISSUE TITLE
Reboot Core causes crash with relative paths

### ISSUE THREAD

**Issue by Meerkov** [TASVideos/BizHawk#2077]
Version 2.4.2
Windows 10

1) Run this command from the commandline to load a game, and see it loads successfully. Notice the folder structure and relative paths put /Bizhawk/ and /Games/ as siblings in the current directory.
$ ./BizHawk/EmuHawk.exe ./Games/MyGame.nes

2) Reboot core by pressing
CTRL+R

3) Notice the game doesn't reboot. Try again
CTRL+R

4) EmuHawk crashes.
```
System.ArgumentNullException: Value cannot be null.
Parameter name: path
   at BizHawk.Client.EmuHawk.MainForm.LoadRomInternal(String path, LoadRomArgs args)
   at BizHawk.Client.EmuHawk.MainForm.LoadRom(String path, LoadRomArgs args)
   at BizHawk.Client.EmuHawk.MainForm.CheckHotkey(String trigger)
   at BizHawk.Client.EmuHawk.MainForm.<ProcessInput>b__101_0(Boolean current, String trigger)
   at System.Linq.Enumerable.Aggregate[TSource,TAccumulate](IEnumerable`1 source, TAccumulate seed, Func`3 func)
   at BizHawk.Client.EmuHawk.MainForm.ProcessInput()
   at BizHawk.Client.EmuHawk.MainForm.ProgramRunLoop()
   at BizHawk.Client.EmuHawk.Program.SubMain(String[] args)
```

5) Repeat step 1 and 2.
6) Notice that the game file in "Recent Games is "./Games/MyGame.nes"
7) Try to load that but you get:
```
Could not open ./Games/MyGame.nes  
Remove from list?
```
Conclusion:
When it loads the first time, it correctly is identifying that the filepath is relative to the parent folder of /Bizhawk/. However, when it tries to load in step 6, it probably is trying to load ./BizHawk/Games/MyGame.nes instead, and failing. Not sure why it crashes in step 3.

**Comment by zeromus** [comment]
works fine for me in windows 7. maybe try not using bash.

**Comment by Meerkov** [comment]
This is using command prompt. Just to confirm, here isone version of the command I was running, with a lua parameter, if that matters.

.\BizHawk\EmuHawk.exe ./Games/MyGame.nes --lua=..\Scripts\myScript.lua

**Comment by Meerkov** [comment]
I also get this error, presumably depending on where in my Lua script it has reached when the game fails to load.

NLua.Exceptions.LuaException: unprotected error in call to Lua API (0)
   at NLua.Lua.PanicCallback(IntPtr luaState)
   at lua_error(lua_State* )
   at NLua.ObjectTranslator.throwError(IntPtr luaState, Object e)
   at NLua.Lua.SetPendingException(Exception e)
   at NLua.LuaMethodWrapper.call(IntPtr luaState)
   at NLua.MetaFunctions.runFunctionDelegate(IntPtr luaState)
   at lua_resume(lua_State* , Int32 )
   at BizHawk.Client.EmuHawk.Win32LuaLibraries.ResumeScript(LuaFile lf)
   at BizHawk.Client.EmuHawk.LuaConsole.<>c__DisplayClass60_1.<ResumeScripts>b__1()
   at BizHawk.Client.Common.LuaSandbox.Sandbox(Action callback, Action exceptionCallback)

Additionally, the script in the lua console is "..\Scripts\MyScript.lua" regardless of if the file exists. Upon reloading, this can cause a the file-not-found to throw a loud error (but it only hits the lua console on startup).

**Comment by zeromus** [comment]
the problem is the lua. something about rebooting the core is leaving lua's current directory management confused, which leaves the process's current directory confused. this is strictly a lua bug

**Comment by YoshiRulz** [comment]
Partially fixed in 7a7522f28. I spent so long getting the rom part to work that I didn't even get to the Lua script part. I did get *a* crash when testing different arguments, but I got fed up with PowerShell and didn't bother trying again after that fix.
